diff --git a/include/openssl/base.h b/include/openssl/base.h
index 95cc15df1..2d5dbcb02 100644
--- a/include/openssl/base.h
+++ b/include/openssl/base.h
@@ -414,10 +414,10 @@ typedef void *OPENSSL_BLOCK;

 #if !defined(BORINGSSL_NO_CXX)

-extern "C++" {
-
 #include <memory>

+extern "C++" {
+
 // STLPort, used by some Android consumers, not have std::unique_ptr.
 #if defined(_STLPORT_VERSION)
 #define BORINGSSL_NO_CXX
 