diff --git a/include/grpc/support/port_platform.h b/include/grpc/support/port_platform.h
index 04a79ef3c8..6db7822131 100644
--- a/include/grpc/support/port_platform.h
+++ b/include/grpc/support/port_platform.h
@@ -30,7 +30,7 @@
  * Defines GPR_ABSEIL_SYNC to use synchronization features from Abseil
  */
 #ifndef GPR_ABSEIL_SYNC
-#if defined(__APPLE__)
+#if defined(__APPLE__) || defined(GRPC_NO_ABSL_SYNC)
 // This is disabled on Apple platforms because macos/grpc_basictests_c_cpp
 // fails with this. https://github.com/grpc/grpc/issues/23661
 #else
